File: dp_labs_and_solutions/dp_labs_and_solutions/dp_matlab_labs_and_solutions/interpreter/solution/Literal.m
% Literal.m

classdef Literal < Terminal
	properties (GetAccess = public, SetAccess = private)
		value % int
	end

	methods
		% constructor:
		function theLiteral = Literal(aValue)
			theLiteral.value = aValue;
		end

		function result = getNumberOfUsesOfVariable(theLiteral, aName)
			result = 0;
		end

		function result = getValue(theLiteral, aMap)
			result = theLiteral.value;
		end

		function output(theLiteral)
			fprintf('%d', theLiteral.value);
		end
	end
end

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/interpreter/lab/literal.cpp
// literal.cpp

#include <iostream>
#include <string>
class expression;
#include "literal.h"

literal::literal(int a_value)
{
	set_value(a_value);
}

int literal::get_number_of_uses_of_variable(const std::string & a_name) const
{
	int result(0);

	return result;
}

void literal::output(std::ostream & a_ostream) const
{
	a_ostream << get_value();
}

void literal::set_value(int a_value)
{
	my_value = a_value;
}

int literal::get_value() const
{
	return my_value;
}
